### DISCUSSION TITLE
Format code to std out

### DISCUSSION THREAD

**Discussion by thomasheartman** [biomejs/biome#513] [Q&A] [answered]
Hi! 

I was looking through the docs for the `format` command and wanted to know if it's possible to write the formatted content to stdout? So if I format a file, I'll get the file's content formatted to std out? I couldn't find the flag to do that, but am thinking that I might be missing something.

Edit: just to be clear: I know that if you skip the `--write` flag, you'll get a summary of what lines will be changed written to std out, but I'd like the fully formatted file.

The reason for me asking is I'm trying to integrate this with Emacs. I found a package ([emacs-format-all](https://github.com/lassik/emacs-format-all-the-code/)) that I should be able to modify pretty easily to use biome, but it requires that the new content is written to std out. Of course, there are other ways to integrate with biome (like using a standard before-save hook), but this seemed like an easy way to go. Happy to take advice if you have any ☺️

**Comment by ematipico** [comment, marked as answer]
You probably want to use `--stdin-file-path`. From the documentation:

> --stdin-file-path=PATH — A file name with its extension to pass when reading from standard in, e.g. echo ‘let a;’ | biome format —stdin-file-path=file.js”.

By doing so, Biome will print the result to `stdout`

**Reply by thomasheartman** [reply]
Ah, interesting! I saw that option, but I couldn't quite understand what it was doing. What does it do? The docs say:

> A file name with its extension to pass when reading from standard in, e.g. echo ‘let a;’ | biome check —stdin-file-path=file.js”

But I don't understand what that means. If it's printing to std out, why does it need a file path?

**Reply by ematipico** [reply]
Biome uses the extension of the file to determine how to format the content. The file doesn't need to exist, it can be any name. What's important is the extension

**Reply by thomasheartman** [reply]
Ah, okay, that makes sense; thanks! So the only way to make it print to std out is to pipe data into it, yeah?

**Reply by ematipico** [reply]
Yes, exactly
